app/services/foreman_templates/template_importer.rb

```diff
@@ -33,94 +34,97 @@ def import_from_git
 
       begin
         gitrepo = Git.clone(@repo, @dir)
-        branch = @branch ? @branch : get_default_branch(gitrepo)
-        gitrepo.checkout(branch) if branch
+        gitrepo.checkout(@branch) if @branch
```

Comment:
If @branch is loaded from settings and it's empty string, this will fail I'm afraid. Since we dropped get_default_branch we should either add `@branch.blank?` or fallback to "master"

and I think we should reuse `TemplateExporter#setup_git_branch` so perhaps move it to action and reuse here?